--- AgileX Base
+++ LCAS expanded
@@ -1,252 +1,233 @@
-<robot name="hunter2_base">
-  <link name="base_link">
+<robot name="hunter">
+  <link name="base_link" />
+  <joint name="chassis_joint" type="fixed">
+    <parent link="base_link" />
+    <child link="chassis" />
+    <origin rpy="0 0 0" xyz="0 0 0.37" />
+  </joint>
+  <link name="chassis">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/chassis.dae" />
+      </geometry>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/chassis.dae" />
+      </geometry>
+    </collision>
     <inertial>
       <origin xyz="0.0266989327622036 0.000409203500587652 -0.135901124750884" rpy="0 0 0" />
-      <mass value="6.02945169536679" />
+      <mass value="72" />
       <inertia ixx="0.0332370959804736" ixy="-1.013150489971E-06" ixz="-0.00188859201421112" iyy="0.111072002332437" iyz="-5.13308150598312E-07" izz="0.12660862809283" />
     </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/base_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
+  </link>
+  <link name="rear_left_wheel">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/rear_left_wheel.dae" />
+      </geometry>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/rear_left_wheel.dae" />
+      </geometry>
+    </collision>
+    <inertial>
+      <origin xyz="-6.5774E-06 2.0242E-05 -0.0096386" rpy="0 0 0" />
+      <mass value="7.0" />
+      <inertia ixx="0.063724" ixy="0" ixz="0" iyy="0.063724" iyz="0" izz="0.11072" />
+    </inertial>
+  </link>
+  <joint name="rear_left_wheel_joint" type="continuous">
+    <parent link="chassis" />
+    <child link="rear_left_wheel" />
+    <origin xyz="-0.28 0.2925 -0.19558" rpy="1.5707963267948966 0 0" />
+    <axis xyz="0 0 -1" />
+    <dynamics damping="0.2" friction="15" />
+    <limit effort="10.0" velocity="10.0" />
+  </joint>
+  <gazebo reference="rear_left_wheel">
+    <mu1 value="1.0" />
+    <mu2 value="0.5" />
+    <kp value="10000000.0" />
+    <kd value="1.0" />
+    <fdir1 value="1 0 0" />
+  </gazebo>
+  <link name="rear_right_wheel">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/rear_right_wheel.dae" />
+      </geometry>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/rear_right_wheel.dae" />
+      </geometry>
+    </collision>
+    <inertial>
+      <origin xyz="-6.5774E-06 2.0242E-05 -0.0096386" rpy="0 0 0" />
+      <mass value="7.0" />
+      <inertia ixx="0.063724" ixy="0" ixz="0" iyy="0.063724" iyz="0" izz="0.11072" />
+    </inertial>
+  </link>
+  <joint name="rear_right_wheel_joint" type="continuous">
+    <parent link="chassis" />
+    <child link="rear_right_wheel" />
+    <origin xyz="-0.28 -0.2925 -0.19558" rpy="1.5707963267948966 0 0" />
+    <axis xyz="0 0 -1" />
+    <dynamics damping="0.2" friction="15" />
+    <limit effort="10.0" velocity="10.0" />
+    <mimic joint="rear_left_wheel_joint" multiplier="1.0" offset="0.0" />
+  </joint>
+  <gazebo reference="rear_right_wheel">
+    <mu1 value="1.0" />
+    <mu2 value="0.5" />
+    <kp value="10000000.0" />
+    <kd value="1.0" />
+    <fdir1 value="1 0 0" />
+  </gazebo>
+  <link name="front_left_steering_link">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <cylinder length="0.01" radius="0.01" />
+      </geometry>
+      <material name="invisible">
+        <color rgba="0.0 0.0 0.0 0.0" />
       </material>
     </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/base_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <link name="front_steer_left_link">
-    <inertial>
-      <origin xyz="-0.012614 1.8116E-05 0.0023132" rpy="0 0 0" />
-      <mass value="7.8526" />
-      <inertia ixx="0.063827" ixy="1.5154E-07" ixz="3.1921E-07" iyy="0.063827" iyz="2.9854E-08" izz="0.11091" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_steer_left_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.79216 0.81961 0.93333 1" />
+    <inertial>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <mass value="0.005" />
+      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1" />
+    </inertial>
+  </link>
+  <joint name="front_left_steering_joint" type="revolute">
+    <parent link="chassis" />
+    <child link="front_left_steering_link" />
+    <origin xyz="0.37142 0.2925 -0.19558" rpy="1.5707963267948966 0 0" />
+    <axis xyz="0 1 0" />
+    <limit lower="-0.461" upper="0.461" effort="10.0" velocity="1" />
+    <dynamics damping="0.2" />
+  </joint>
+  <link name="front_left_wheel">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/front_left_wheel.dae" />
+      </geometry>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/front_left_wheel.dae" />
+      </geometry>
+    </collision>
+    <inertial>
+      <origin xyz="-6.5774E-06 2.0242E-05 -0.0096386" rpy="0 0 0" />
+      <mass value="7.0" />
+      <inertia ixx="0.063724" ixy="0" ixz="0" iyy="0.063724" iyz="0" izz="0.11072" />
+    </inertial>
+  </link>
+  <joint name="front_left_wheel_joint" type="continuous">
+    <parent link="front_left_steering_link" />
+    <child link="front_left_wheel" />
+    <origin xyz="0 0 0" rpy="0 0 0" />
+    <axis xyz="0 0 -1" />
+  </joint>
+  <link name="front_right_steering_link">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <cylinder length="0.01" radius="0.01" />
+      </geometry>
+      <material name="invisible">
+        <color rgba="0.0 0.0 0.0 0.0" />
       </material>
     </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_steer_left_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="front_steer_left_joint" type="revolute">
-    <origin xyz="0.37142 0.29199 -0.1955" rpy="1.5708 0 0" />
-    <parent link="base_link" />
-    <child link="front_steer_left_link" />
-    <axis xyz="0 0 1" />
-    <limit lower="0" upper="0" effort="100" velocity="1" />
-  </joint>
-  <link name="front_left_wheel_link">
-    <inertial>
-      <origin xyz="-6.5774E-06 2.0242E-05 -0.0096386" rpy="0 0 0" />
-      <mass value="7.7007" />
-      <inertia ixx="0.063724" ixy="-3.8745E-08" ixz="2.2753E-07" iyy="0.063724" iyz="-4.88E-08" izz="0.11072" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_left_wheel_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_left_wheel_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="front_left_wheel_joint" type="continuous">
+    <inertial>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <mass value="0.005" />
+      <inertia ixx="0.1" ixy="0" ixz="0" iyy="0.1" iyz="0" izz="0.1" />
+    </inertial>
+  </link>
+  <joint name="front_right_steering_joint" type="revolute">
+    <parent link="chassis" />
+    <child link="front_right_steering_link" />
+    <origin xyz="0.37142  -0.2925 -0.19558" rpy="1.5707963267948966 0 0" />
+    <axis xyz="0 1 0" />
+    <limit lower="-0.461" upper="0.461" effort="10.0" velocity="1" />
+    <dynamics damping="0.2" />
+    <mimic joint="front_left_steering_joint" multiplier="1.0" offset="0.0" />
+  </joint>
+  <link name="front_right_wheel">
+    <visual>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/front_right_wheel.dae" />
+      </geometry>
+    </visual>
+    <collision>
+      <origin xyz="0 0 0" rpy="0 0 0" />
+      <geometry>
+        <mesh filename="file://third_party/hunter_robot/hunter_description/meshes/front_right_wheel.dae" />
+      </geometry>
+    </collision>
+    <inertial>
+      <origin xyz="-6.5774E-06 2.0242E-05 -0.0096386" rpy="0 0 0" />
+      <mass value="7.0" />
+      <inertia ixx="0.063724" ixy="0" ixz="0" iyy="0.063724" iyz="0" izz="0.11072" />
+    </inertial>
+  </link>
+  <joint name="front_right_wheel_joint" type="continuous">
+    <parent link="front_right_steering_link" />
+    <child link="front_right_wheel" />
     <origin xyz="0 0 0" rpy="0 0 0" />
-    <parent link="front_steer_left_link" />
-    <child link="front_left_wheel_link" />
-    <axis xyz="0 0 1" />
-  </joint>
-  <link name="front_steer_right_link">
-    <inertial>
-      <origin xyz="-8.9374E-08 1.0997E-07 0.0095958" rpy="0 0 0" />
-      <mass value="7.696" />
-      <inertia ixx="0.063704" ixy="-6.3892E-08" ixz="-2.0159E-08" iyy="0.063704" iyz="6.4721E-08" izz="0.11068" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_steer_right_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_steer_right_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="front_steer_right_joint" type="revolute">
-    <origin xyz="0.37142 -0.29201 -0.1955" rpy="1.5708 0 0" />
-    <parent link="base_link" />
-    <child link="front_steer_right_link" />
-    <axis xyz="0 0 -1" />
-    <limit lower="0" upper="0" effort="100" velocity="1" />
-  </joint>
-  <link name="front_right_wheel_link">
-    <inertial>
-      <origin xyz="-9.1148E-08 1.1284E-07 0.0095049" rpy="0 0 0" />
-      <mass value="7.5613" />
-      <inertia ixx="0.063626" ixy="-6.4092E-08" ixz="-2.0164E-08" iyy="0.063626" iyz="6.4721E-08" izz="0.11054" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_right_wheel_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_right_wheel_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="front_right_wheel_joint" type="continuous">
-    <origin xyz="0 0 0" rpy="0 0 0" />
-    <parent link="front_steer_right_link" />
-    <child link="front_right_wheel_link" />
-    <axis xyz="0 0 -1" />
-  </joint>
-  <link name="left_rear_link">
-    <inertial>
-      <origin xyz="-8.826E-05 6.4342E-05 -0.0096491" rpy="0 0 0" />
-      <mass value="7.6692" />
-      <inertia ixx="0.063699" ixy="-2.542E-07" ixz="2.638E-08" iyy="0.063699" iyz="-5.9934E-08" izz="0.11067" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/left_rear_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/left_rear_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="left_rear_joint" type="continuous">
-    <origin xyz="-0.28 0.29249 -0.19558" rpy="1.5708 0 0" />
-    <parent link="base_link" />
-    <child link="left_rear_link" />
-    <axis xyz="0 0 1" />
-    <limit effort="100" velocity="1" />
-  </joint>
-  <link name="right_rear_link">
-    <inertial>
-      <origin xyz="8.6344E-05 7.4488E-05 0.0097824" rpy="0 0 0" />
-      <mass value="7.7133" />
-      <inertia ixx="0.06373" ixy="2.5408E-07" ixz="2.6379E-08" iyy="0.06373" iyz="5.9941E-08" izz="0.11073" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/right_rear_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/right_rear_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="right_rear_joint" type="continuous">
-    <origin xyz="-0.28018 -0.29251 -0.19558" rpy="1.5708 0.0026575 0" />
-    <parent link="base_link" />
-    <child link="right_rear_link" />
-    <axis xyz="0 0 -1" />
-    <limit effort="100" velocity="1" />
-  </joint>
-  <link name="front_steer_link">
-    <inertial>
-      <origin xyz="0.049836 8.1046E-15 0.017912" rpy="0 0 0" />
-      <mass value="0.0049179" />
-      <inertia ixx="1.6846E-07" ixy="7.6396E-23" ixz="-1.0974E-08" iyy="7.549E-07" iyz="-1.8957E-22" izz="7.334E-07" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_steer_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/front_steer_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="front_steer_joint" type="fixed">
-    <origin xyz="0.44792 -1.1431E-05 -0.121" rpy="0 0 0" />
-    <parent link="base_link" />
-    <child link="front_steer_link" />
-    <axis xyz="0 0 0" />
-  </joint>
-  <link name="rear_wheel_link">
-    <inertial>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <mass value="0" />
-      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0" />
-    </inertial>
-    <visual>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/rear_wheel_link.STL" />
-      </geometry>
-      <material name="">
-        <color rgba="1 1 1 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin xyz="0 0 0" rpy="0 0 0" />
-      <geometry>
-        <mesh filename="package://hunter2_base/meshes/rear_wheel_link.STL" />
-      </geometry>
-    </collision>
-  </link>
-  <joint name="rear_wheel_joint" type="fixed">
-    <origin xyz="-0.42408 -1.1431E-05 -0.085581" rpy="0.0032313 0 0" />
-    <parent link="base_link" />
-    <child link="rear_wheel_link" />
-    <axis xyz="0 0 0" />
-  </joint>
+    <axis xyz="0 0 -1" />
+  </joint>
+  <ros2_control name="GazeboSystem" type="system">
+    <hardware>
+      <plugin>gazebo_ros2_control/GazeboSystem</plugin>
+    </hardware>
+    <joint name="front_left_steering_joint">
+      <command_interface name="position" />
+      <state_interface name="position" />
+    </joint>
+    <joint name="front_right_steering_joint">
+      <param name="mimic">front_left_steering_joint</param>
+      <param name="multiplier">1</param>
+      <state_interface name="position" />
+    </joint>
+    <joint name="rear_left_wheel_joint">
+      <command_interface name="velocity" />
+      <state_interface name="velocity" />
+      <state_interface name="position" />
+    </joint>
+    <joint name="rear_right_wheel_joint">
+      <param name="mimic">rear_left_wheel_joint</param>
+      <param name="multiplier">1</param>
+      <state_interface name="velocity" />
+      <state_interface name="position" />
+    </joint>
+    <joint name="front_right_wheel_joint">
+      <state_interface name="velocity" />
+      <state_interface name="position" />
+    </joint>
+    <joint name="front_left_wheel_joint">
+      <state_interface name="velocity" />
+      <state_interface name="position" />
+    </joint>
+  </ros2_control>
+  <gazebo>
+    <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
+      <parameters>third_party/hunter_robot/hunter_description/config/ackermann_like_controller.yaml</parameters>
+    </plugin>
+  </gazebo>
 </robot>
