--- before/hunter2_sim.urdf
+++ after/hunter2_sim.urdf
@@ -60,29 +60,6 @@
         iyz="2.9854E-08"
         izz="0.11091" />
     </inertial>
-    <visual>
-      <origin
-        xyz="0 0 0"
-        rpy="0 0 0" />
-      <geometry>
-        <mesh
-          filename="meshes/front_steer_left_link.STL" />
-      </geometry>
-      <material
-        name="">
-        <color
-          rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin
-        xyz="0 0 0"
-        rpy="0 0 0" />
-      <geometry>
-        <mesh
-          filename="meshes/front_steer_left_link.STL" />
-      </geometry>
-    </collision>
   </link>
   <joint
     name="front_steer_left_joint"
@@ -95,7 +72,7 @@
     <child
       link="front_steer_left_link" />
     <axis
-      xyz="0 0 1" />
+      xyz="0 1 0" />
     <limit
       lower="-0.58"
       upper="0.58"
@@ -171,29 +148,6 @@
         iyz="6.4721E-08"
         izz="0.11068" />
     </inertial>
-    <visual>
-      <origin
-        xyz="0 0 0"
-        rpy="0 0 0" />
-      <geometry>
-        <mesh
-          filename="meshes/front_steer_right_link.STL" />
-      </geometry>
-      <material
-        name="">
-        <color
-          rgba="0.79216 0.81961 0.93333 1" />
-      </material>
-    </visual>
-    <collision>
-      <origin
-        xyz="0 0 0"
-        rpy="0 0 0" />
-      <geometry>
-        <mesh
-          filename="meshes/front_steer_right_link.STL" />
-      </geometry>
-    </collision>
   </link>
   <joint
     name="front_steer_right_joint"
@@ -206,7 +160,7 @@
     <child
       link="front_steer_right_link" />
     <axis
-      xyz="0 0 -1" />
+      xyz="0 1 0" />
     <limit
       lower="-0.58"
       upper="0.58"
